packages/ngtools/webpack/src/resource_loader.ts

```diff
@@ -39,11 +40,15 @@ export class WebpackResourceLoader {
   }
 
   getResourceDependencies(filePath: string) {
-    return this._fileDependencies.get(filePath) || [];
+    const set = this._fileDependencies.get(filePath);
+
+    return set ? Array.from(set.values()) : [];
   }
 
   getAffectedResources(file: string) {
-    return this._reverseDependencies.get(file) || [];
+    const set = this._reverseDependencies.get(file);
+
+    return set ? Array.from(set.values()) : [];
```

Comment:
nit: This can return the Set directly.  Based on usage the return type is `Iterable<string>`.  This would prevent the collection from being cloned on each call.

`getResourceDependencies` should ideally be the same since the results appear to only be iterated but would require small adjustments to the iteration usage (`for..of` instead of `.foreach()`).

Reply:
Added, but left `getDependencies` still pushing the elements to array since that method is gathering a lot of different arrays.

Reply:
looks good.

When this makes it in, it should greatly simplify iterator usage: https://github.com/tc39/proposal-iterator-helpers